funcx_sdk/funcx/sdk/container_spec.py

```diff
@@ -1,18 +1,52 @@
+import re
+from typing import List, Optional
+
+
 class ContainerSpec:
+    # RegEx copied from https://ihateregex.io/expr/semver/
+    semver_regex = re.compile(
+        r"^(0|[1-9]\d*)\.(0|[1-9]\d*)(\.(0|[1-9]\d*))?(?:-((?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*)(?:\.(?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*))*))?(?:\+([0-9a-zA-Z-]+(?:\.[0-9a-zA-Z-]+)*))?$"  # noqa E501
+    )
+
     def __init__(
         self,
-        name=None,
-        description=None,
-        apt=None,
-        pip=None,
-        conda=None,
-        payload_url=None,
+        name: Optional[str] = None,
+        description: Optional[str] = None,
+        apt: Optional[List[str]] = None,
+        pip: Optional[List[str]] = None,
+        conda: Optional[List[str]] = None,
+        python_version: str = "3.7",  # This is the default for repo2docker
+        payload_url: Optional[str] = None,
```

Comment:
As of late, we've been slowly migrating towards the "or" (`|`) syntax, where feasible.  So for these, this would be:
```python
from __future__ import annotations
...
name: str | None = None,
description: str | None = None,
apt: list[str] | None = None,
```

Reply:
Showing my Scala roots...